src/solc_0.7/GameToken/GameToken.sol

```diff
@@ -0,0 +1,429 @@
+        _asset = asset;
+        _minter = initialMinter;
+    }
+
+    ///////////////////////////////  Modifiers //////////////////////////////
+
+    modifier notToZero(address to) {
+        require(to != address(0), "DESTINATION_ZERO_ADDRESS");
+        _;
+    }
+
+    modifier notToThis(address to) {
+        require(to != address(this), "DESTINATION_GAME_CONTRACT");
+        _;
+    }
+
+    ///////////////////////////////  Functions //////////////////////////////
+
+    /// @notice Function to get the amount of each assetId in a GAME
+    /// @param gameId The game to query
+    /// @param assetIds The assets to get balances for
+    function getAssetBalances(uint256 gameId, uint256[] calldata assetIds)
+        external
+        view
+        override
+        returns (uint256[] memory)
+    {
+        uint256 length = assetIds.length;
+        uint256[] memory assets;
+        assets = new uint256[](length);
+        for (uint256 i = 0; i < length; i++) {
+            assets[i] = _gameAssets[gameId][assetIds[i]];
+        }
+        return assets;
+    }
+
+    /// @notice Function to allow token owner to set game editors
+    /// @param from The address of the one creating the game (may be different from msg.sender if metaTx)
+    /// @param gameId The id of the GAME token owned by owner
+    /// @param editor The address of the editor to set
+    /// @param isEditor Add or remove the ability to edit
+    function setGameEditor(
+        address from,
+        uint256 gameId,
+        address editor,
+        bool isEditor
+    ) external override {
+        require(msg.sender == _ownerOf(gameId) || _isValidMetaTx(from), "EDITOR_ACCESS_DENIED");
+        _setGameEditor(gameId, editor, isEditor);
+    }
+
+    /// @dev Function to allow token owner to set game editors
+    /// @param gameId The id of the GAME token owned by owner
+    /// @param editor The address of the editor to set
+    /// @param isEditor Add or remove the ability to edit
+    function _setGameEditor(
+        uint256 gameId,
+        address editor,
+        bool isEditor
+    ) internal {
+        emit GameEditorSet(gameId, editor, isEditor);
+        _gameEditors[gameId][editor] = isEditor;
+    }
+
+    /// @notice Transfers creatorship of `original` from `sender` to `to`.
+    /// @param sender address of current registered creator.
+    /// @param original address of the original creator whose creation are saved in the ids themselves.
+    /// @param to address which will be given creatorship for all tokens originally minted by `original`.
+    function transferCreatorship(
+        address sender,
+        address original,
+        address to
+    ) external override notToZero(to) {
+        require(
+            msg.sender == sender || _isValidMetaTx(sender) || _superOperators[msg.sender],
+            "TRANSFER_ACCESS_DENIED"
+        );
+        require(sender != address(0), "ZERO_SENDER_FORBIDDEN");
```

Comment:
in the other PR that change the error message, we use a different convention for invalid address.
we should pick one